src/mem_scan.c

```diff
@@ -67,7 +67,12 @@
  */
 void scan_mem(sqlite3 *db, char* arg_pid) 
 {
-	int pid = atoi(arg_pid);
+	char *pEnd;
+	long int pid = strtol(arg_pid, &pEnd, 10);
```

Comment:
I think this string paring should be done in scan_mem's caller (main). That would let you avoid the printf churn.

Reply:
Addressed.